src/features/validators/data/getGoodAddresses.ts

```diff
@@ -0,0 +1,13 @@
+// results is an array of booleans that correspond to the groupAddresses block status
+// good: because we want the non blocked groups good is false, but true for getting health groups.
+export function getGoodAddresses(results: boolean[], groupAddresses: string[], good = false) {
```

Comment:
Couldn't this be rewritten to be a bit simpler?

`return groupAddresses.filter((_, index) => result[index] === good)` . Also IMHO `good` is confusing but you explained it about so that's ok! 👍

Reply:
I like having it return a Set. and not need an intermediate Array to do so.